### PR TITLE
feat: dashboard filters simplified view mode

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/6478

### Description:

- [x] Removed "tiles" configuration tab
- [x] Removed tabs from top
- [x] Removed filter label
- [x] If default values were entered by editor, show them in the widget
- [x] If default values were not entered, show Placeholder text in the widget (but `any value` in the pill)
- [x] Reset action

CleanShot 2023-08-04 at 20 57 47@2x

reset to original value:
CleanShot 2023-08-07 at 16 40 03@2x

### CODE REVIEW

**Comment by IrakliJani** [comment]
p.s. @PriPatel I don't know why but I picked this button.
we also have these options:

CleanShot 2023-08-07 at 16 54 21@2x

wdyt?

**Comment by PriPatel** [comment]
@IrakliJani thanks for this!
the icon you have chosen already works best :)

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx [hunk 1]
```diff
@@ -138,81 +164,95 @@ const FilterConfiguration: FC<Props> = ({
...
+                value={selectedTabId}
+                onTabChange={(tabId: FilterTabs) => onTabChange(tabId)}
             >
-                <Tab
-                    id="settings"
-                    title={
-                        <Tooltip2
-                            content="Select the value you want to filter your dimension by"
-                            position="bottom"
+                {isEditMode || !isActiveFilter ? (
```
Comment (IrakliJani) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:176]: remove isActiveFilter

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx [hunk 2]
```diff
@@ -81,6 +82,31 @@ const FilterConfiguration: FC<Props> = ({
                   ),
         );
 
+    const isFilterModified = useMemo(() => {
+        if (!originalFilterRule || !originalFilterRule) return false;
```
Comment (IrakliJani) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:86]: remove one of them

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx [hunk 3]
```diff
@@ -81,6 +82,31 @@ const FilterConfiguration: FC<Props> = ({
...
+
+        // fix serialization of date fields
+        const serializedInternalFilterRule = produce(
+            internalFilterRule,
+            (draft) => {
+                if (draft.values && draft.values.length > 0) {
+                    draft.values = draft.values.map((v) => v.toString());
```
Comment (IrakliJani) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:93]: check date type first and then convert

File: packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx [hunk 4]
```diff
@@ -83,6 +83,12 @@ const DateFilterInputs = <T extends ConditionalRule = DateFilterRule>(
                             <MonthAndYearInput
                                 disabled={disabled}
                                 placeholder={placeholder}
+                                popoverProps={{
+                                    onOpen: () =>
+                                        popoverProps?.onOpened?.(null as any),
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx:88]: add comment why we are doing this

File: packages/frontend/src/components/common/MonthAndYearInput.tsx [hunk 5]
```diff
@@ -1,35 +1,44 @@
...
             size="xs"
+            onClick={toggle}
+            {...props}
             popoverProps={{
                 withArrow: true,
                 withinPortal: false,
                 shadow: 'md',
+                ...props.popoverProps,
+                opened: isPopoverOpen,
+                onOpen: () => {
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/MonthAndYearInput.tsx:28]: link issue related to mantine here as a comment

**Comment by joaoviana** [comment]
Create dashboard filter with empty value; 
Save
Click on filter in view mode
`Reset` button should be disabled: 
image

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx [hunk 6]
```diff
@@ -138,81 +164,95 @@ const FilterConfiguration: FC<Props> = ({
...
+                            variant="default"
+                            color="gray"
+                            onClick={() => {
+                                if (!originalFilterRule) return;
+                                handleChangeFilterRule(originalFilterRule);
+                            }}
+                        >
+                            <MantineIcon icon={IconRotate2} />
+                        </Button>
+                    </Tooltip>
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:239]: btw, it also resets tile settings to default (no matter if we show button there or not.
so the question is, should we be reverting tile config changes with this button or not? it kinda seems useful but it's controversial 

https://github.com/lightdash/lightdash/assets/962095/bde6f007-0c5b-45b6-babc-4d9f5d4d8ede

**Review by joaoviana — APPROVED** [review]

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (PriPatel) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx]: @IrakliJani The button should only be for the values. The tile config is interesting but maybe we shouldn't go there yet! 
But also... the viewer shouldn't be seeing the tiles tab anyway right? So they would never be in a position to 'revert tile config?'

**Pull request merged**